prisma/schema.prisma

```diff
@@ -11,8 +11,28 @@ model User {
   id        String   @id @default(uuid()) @db.Uuid
   createdAt DateTime @default(now()) @map("created_at")
   updatedAt DateTime @updatedAt @map("updated_at")
+  email     String   @unique
+  name      String
+  password  String
+}
 
-  email    String @unique
-  name     String
-  password String
+model SteamUser {
+  id                       String   @id @default(uuid()) @db.Uuid
+  createdAt                DateTime @default(now()) @map("created_at")
+  updatedAt                DateTime @updatedAt @map("updated_at")
+  steamid                  String   @unique
+  communityvisibilitystate Int
+  profilestate             Int
+  personaname              String
+  profileurl               String
+  avatar                   String
+  avatarmedium             String
+  avatarfull               String
+  avatarhash               String
+  lastlogoff               Int
+  personastate             Int
+  realname                 String
+  primaryclanid            String
+  timecreated              Int
+  personastateflags        Int
 }
```

Comment:
Hmm, I don't think you really need all of this data. I'd say that you should keep only `steamid`, `personename`, `profileurl` and `avatar`. The idea of `Steam-auth` is to still use the same user model, but get initial data from Steam + save some information about the user, so we can login him again using that information.

Based on that I'd keep the following model:
```prisma
model User {
  id        String   @id @default(uuid()) @db.Uuid
  createdAt DateTime @default(now()) @map("created_at")
  updatedAt DateTime @updatedAt @map("updated_at")
  email     String   @unique
  name      String
  password  String

  steamUser SteamUser?
}

model SteamUser {
  steamid     String @unique
  personaname String
  profileurl  String
  avatar      String

  userId String @unique @db.Uuid
  user   User   @relation(fields: [userId], references: [id])
}
```

Notice, that `SteamUser` always have a reference to a `User` model, but not every `User` have a connected `SteamUser` model. So when anybody connects an account to steam, you create a `SteamUser` record.

When somebody tries to login using `Steam`, you look for a record of a `SteamUser`, then you can easily find a related `User` and authorize that user